clarifai/runners/models/openai_class.py

```diff
@@ -1,143 +1,222 @@
-                    model=model, messages=messages, temperature=temperature, max_tokens=max_tokens
-                )
+                completion = self._process_request(**params)
+                if completion.get('usage'):
+                    if completion['usage'].get('prompt_tokens') and completion['usage'].get(
+                        'completion_tokens'
+                    ):
+                        self.set_output_context(
+                            prompt_tokens=completion['usage']['prompt_tokens'],
+                            completion_tokens=completion['usage']['completion_tokens'],
+                        )
 
                 return json.dumps(completion)
+
         except Exception as e:
-            error_response = {
-                "error": {
-                    "message": str(e),
-                    "type": "InvalidRequestError",
-                    "code": "invalid_request_error",
-                }
-            }
-            return json.dumps(error_response)
+            return self._format_error_response(e)
 
     @ModelClass.method
-    def openai_stream_transport(self, req: str) -> Iterator[str]:
+    def openai_stream_transport(self, msg: str) -> Iterator[str]:
         """Process an OpenAI-compatible request and return a streaming response iterator.
-
         This method is used when stream=True and returns an iterator of strings directly,
         without converting to a list or JSON serializing.
 
         Args:
-            req: The request as a JSON string.
+            msg: The request as a JSON string.
 
         Returns:
             Iterator[str]: An iterator yielding text chunks from the streaming response.
         """
-        # Parse the incoming message
-        request_data = json.loads(req)
-
-        # Extract key parameters from the request
-        model = request_data.get("model", "")
-        messages = request_data.get("messages", [])
-        temperature = request_data.get("temperature", 1.0)
-        max_tokens = request_data.get("max_tokens", None)
-
-        # Process the streaming request and return the iterator directly
         try:
-            return self._process_streaming_request(
-                model=model, messages=messages, temperature=temperature, max_tokens=max_tokens
-            )
+            request_data = json.loads(msg)
+            params = self._extract_request_params(request_data)
+            for chunk in self._process_streaming_request(**params):
+                if chunk.get('usage'):
+                    if chunk['usage'].get('prompt_tokens') and chunk['usage'].get(
+                        'completion_tokens'
+                    ):
+                        self.set_output_context(
+                            prompt_tokens=chunk['usage']['prompt_tokens'],
+                            completion_tokens=chunk['usage']['completion_tokens'],
+                        )
+                if (chunk.get('choices') and len(chunk['choices']) > 0) or (
```

Comment:
shouldn't we always just do a json.dumps(chunk) here and not have these iffs? Just let it always return what it has so we're always comopatible.